src/elementary_operators.jl

```diff
@@ -191,6 +192,89 @@ function identity_operator(s::IndsNetworkMap; kwargs...)
   return ITensorNetwork(Op("I"), operator_inds)
 end
 
+" Create an operator bitstring corresponding to the number x"
+function point_to_opsum(s::IndsNetworkMap, x::Number, dim::Int)
+  ttn_op = OpSum()
+  ind_to_ind_value_map = calculate_ind_values(s, x, dim)
+  string_sites = [
+    (ind_to_ind_value_map[only(s[v])] == 1) ? ("Dup", v) : ("Ddn", v) for
+    v in dimension_vertices(s, dim)
+  ]
+  add!(ttn_op, 1.0, (string_sites...)...)
+  return ttn_op
+end
+
+" Create an operator which maps a function to 0 at all points in xs"
+function map_to_zero_operator(
+  s::IndsNetworkMap, xs::Vector, dims::Vector=[1 for _ in xs]; truncate_kwargs...
+)
+  @assert length(unique(dims)) <= 2 # TODO: generalize 
+  ttn_op = OpSum()
+  # build I- ∑_p ∏(bit string p)
+  all_ops = []
+  for (x, dim) in zip(xs, dims)
+    b_op = point_to_opsum(s, x, dim)
+    ttn_op += -1.0 * b_op
+    push!(all_ops, b_op)
+  end
+
+  # if we have (I-∑P_x)(I-∑P_y) then
+  # we have an additional +P_x*P_y term
+  # this is equivelent to ∑ P_{all overlap spots}
+  for i in 1:length(all_ops)
+    for j in (i + 1):length(all_ops)
+      b_op1, b_op2 = all_ops[i], all_ops[j]
+      d1, d2 = dims[i], dims[j]
+      (d1 == d2) && continue
+      ttn_op += Ops.expand(b_op1 * b_op2)
+    end
+  end
+  add!(ttn_op, 1.0, "I", first(dimension_vertices(s, first(dims))))
+  return ttn(ttn_op, indsnetwork(s); truncate_kwargs...)
+end
+
+function map_to_zero_operator(s::IndsNetworkMap, x::Number, dim::Int=1; truncate_kwargs...)
+  return map_to_zero_operator(s, [x], [dim]; truncate_kwargs...)
+end
+
+" Map the points xs in dimension dims of the function f to 0"
+function map_to_zeros(
+  f::ITensorNetworkFunction,
+  xs::Vector,
+  dims::Vector=[1 for _ in xs];
+  cutoff,
+  maxdim,
+  truncate_kwargs...,
+)
+  s = indsnetworkmap(f)
+  zero_op = map_to_zero_operator(s, xs, dims; truncate_kwargs...)
+  return operate(zero_op, f; cutoff, maxdim)
+end
+
+function map_to_zeros(f::ITensorNetworkFunction, x::Number, dim::Int=1; kwargs...)
+  return map_to_zeros(f, [x], [dim]; kwargs...)
+end
+
+""" Create an operator which projects into a constant plane """
+function const_plane_op(s::IndsNetworkMap, xs::Vector, dims::Vector; truncate_kwargs...)
```

Comment:
Hmm, should we name this like `delta_operator` or `delta_kernel` ? This operator is just elevating the delta function `delta_xyz` to an operator right? 
So we could just implement its logic directly with existing functions. Specifically I think

```
function delta_op(s::IndsNetworkMap, xs::Vector, dims::Vector; truncate_kwargs...)
    psi = reduce(+, [delta_xyz(s, [x], [dim]) for (i, x) in enumerate(xs)])
    return operator_proj(psi)
end
```
would do the trick and be quite clean.

Reply:
This function is a different functionality, since the deltas will create a single point but this is making planes

Reply:
If `delta_xyz(s, [x], [dim])` creates the function which is `0` unless `x = x` along the dimension `dim` then this is surely equivalent

Reply:
If it doesn't then we should change the definition of `delta_xyz`

Reply:
Yeah I think you are suggesting an overhaul of what the delta function system does